### PR TITLE
[DTP-460] Convert depositor gRNA sequences to uppercase in count_spacers.py

### PR DESCRIPTION
**Pull request by saksh2311**
https://addgene.atlassian.net/browse/DTP-460

* [x] No [Personally Identifiable Information](https://en.wikipedia.org/wiki/Personally_identifiable_information) is committed.
Reviewer:
* [x] AT Passes

### CODE REVIEW

File: toolkit/parameters.yml
Comment (jasonniehaus) [inline toolkit/parameters.yml:1]: The changes in this file are fine for testing, but should not be committed. We want the repository to reflect the example data files already present

File: workdir/examples/data/count_spacers/metab_gRNA.csv
Comment (jasonniehaus) [inline workdir/examples/data/count_spacers/metab_gRNA.csv:1]: Similarly, this file should not be committed to the repo

File: workdir/examples/data/count_spacers/metab_gRNA_lowercase.csv
Comment (jasonniehaus) [inline workdir/examples/data/count_spacers/metab_gRNA_lowercase.csv:1]: Similarly, this file should not be committed to the repo

**Review by jasonniehaus — COMMENTED** [review]
Please remove changes and input files used for testing

File: toolkit/count_spacers.py [hunk 1]
```diff
@@ -145,9 +145,19 @@ def _process_fastq_file(self, file, library_sequences):
...
+			for rows in reader:
+				try:
+					key = rows[0].upper()
+					# Remove any non-ASCII characters
+					key = ''.join(char for char in key if ord(char) < 128)
+					library_sequences[key] = 0
+				except UnicodeDecodeError:
+					# Handle encoding issues
+					print("Encoding issue with sequence:", rows[0])
+		return library_sequences
```
Comment (bhardwajshagun) [inline toolkit/count_spacers.py]: This looks good, but could removing non-ASCII characters cause any issues @jasonniehaus ?

Comment (jasonniehaus) [reply toolkit/count_spacers.py]: We expect the library sequences to be guide RNA sequences (i.e. a string of A,C,G,T; possible N or other degenerate bases), so removing non-ASCII characters is OK, but not necessarily complete. Our intention is not to catch all possible errors here. @saksh2311 Let's remove this section, as it's not needed for the example files provided in the repo.

Comment (saksh2311) [reply toolkit/count_spacers.py]: okay

**Review by jasonniehaus — APPROVED** [review]
Looks good to me!

**Pull request merged**